underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -106,14 +105,30 @@
   /** Whether the streaming upload is enabled. */
   private final boolean mStreamingUploadEnabled;
 
+//  private final Supplier<ObjectPermissions> mPermissions = this::getPermissionsInternal;
+
   /** The permissions associated with the bucket. Fetched once and assumed to be immutable. */
-  private final Supplier<ObjectPermissions> mPermissions = Suppliers
-      .memoize(new Supplier<ObjectPermissions>() {
-        @Override
-        public ObjectPermissions get() {
-          return getPermissionsInternal();
+  private final Supplier<ObjectPermissions> mPermissions = memoize(this::getPermissionsInternal);
+
+  /** Memoize implementation for java.util.function.supplier. */
+  private static <T> Supplier<T> memoize(Supplier<T> original) {
+    return new Supplier<T>() {
+      Supplier<T> mDelegate = this::firstTime;
```

Comment:
`= firstTime()`?

Reply:
This expression is the method reference version of the following assignment:
``` Supplier<T> mDelegate = () -> firstTime(); ```
And the former one is preferred by IDE .